source/core/material/noise.h

```diff
@@ -175,20 +206,36 @@ const OptimizedNoiseInfo* GetRecommendedOptimizedNoise();
 /// Get a specific noise generator implementation.
 const OptimizedNoiseInfo* GetOptimizedNoise(std::string name);
 
+// Function pointers that will decide the flow of functions
 extern NoiseFunction Noise;
 extern DNoiseFunction DNoise;
+extern NoiseFunction2D Noise2D;
+extern DNoiseFunction2D DNoise2D;
+extern NoiseFunction8D Noise8D;
+extern DTurbulenceFunction DTurbulence;
+extern Initialize_WavesFunction Initialize_Waves;
+extern TurbulenceFunction Turbulence;
 
 void Initialise_NoiseDispatch();
 
 #else // TRY_OPTIMIZED_NOISE
 
 inline DBL Noise(const Vector3d& EPoint, int noise_generator) { return PortableNoise(EPoint, noise_generator); }
 inline void DNoise(Vector3d& result, const Vector3d& EPoint) { PortableDNoise(result, EPoint); }
-
+inline void Noise2D(const Vector3d& EPoint, int noise_generator, double &value) { }//PortableNoise2D(EPoint, noise_generator, value); }
+inline void DNoise2D(Vector3d& result, const Vector3d& EPoint) { }//PortableDNoise2D(result, EPoint); }
+inline DBL Noise8D(Vector3d& result, const Vector3d& EPoint) { return 0.0;}
```

Comment:
The signature of Noise8D does not match its function pointer defined above:

`typedef DBL(*NoiseFunction8D) (const Vector3d& EPoint, int noise_generator);`

This will trigger a build break with the error: no matching function for call to 'Noise8D' on systems where TRY_OPTIMIZED_NOISE is not defined - e.g. macOS.

Cheers,